ibis/expr/operations/window.py

```diff
@@ -71,7 +71,7 @@ class WindowFrame(Value):
     shape = ds.columnar
 
     def __init__(self, start, end, **kwargs):
-        if start and end and start.dtype != end.dtype:
```

Comment:
Could we use `start.dtype.castable(end.dtype)` here instead of introducing `is_same_base_type()`?

Reply:
The logic that we want to implement here is: 
1) when `start` and `end` have the same base type, don't cast even though it's castable (e.g., start=1000, end=0, start is int and end is int8). This is the case that was failing before, because we were using `dt.higher_precendence()` to determine the data type of higher precedence to cast to, and `dt.higher_precendence()` just uses the `castable()` method underneath the hood.
2) else (if `start` and `end` don't have the same base type), if `start` is castable to `end` or `end` is castable to start (e.g., start=ibis.interval(seconds=20), end=0), cast.
3) else (if `start` is not castable to `end` and `end` is not castable to `start`), throw exception.

So case 1 here was not addressed by checking whether the data type is castable.